app/components/navigation_header/index.tsx

```diff
@@ -62,12 +58,17 @@ const NavigationHeader = ({
     const insets = useSafeAreaInsets();
     const styles = getStyleSheet(theme);
 
-    const {largeHeight, defaultHeight} = useHeaderHeight(isLargeTitle, Boolean(subtitle), hasSearch);
+    const {largeHeight, defaultHeight} = useHeaderHeight();
     const containerHeight = useAnimatedStyle(() => {
-        const normal = defaultHeight + insets.top;
-        const calculated = -(insets.top + (scrollValue?.value || 0));
-        return {height: Math.max((normal + calculated), normal)};
-    }, []);
+        const minHeight = defaultHeight + insets.top;
+        const value = -(scrollValue?.value || 0);
+        const height = ((isLargeTitle ? largeHeight : defaultHeight)) + value + insets.top;
+        return {
+            height: Math.max(height, minHeight),
+            minHeight,
+            maxHeight: largeHeight + insets.top + 80,
```

Comment:
That is exactly what we want.
The header will have a minHeight when is collapsed, then the height applies as the header expands/shrinks and finally the maxHeight is used the overscrolling the list (say to pull to refresh) so that the header does not expand all the way.